# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWaylandDisplay.cpp
# Commit: 5f5c42a58f8c
# Full Hash: 5f5c42a58f8cdc3147b92e288ff6a9edef18b899
# Author: stransky <stransky@redhat.com>
# Date: 2025-06-03 21:32:22
# Regressor Bug: 1966639
# File Overlap Count: 1
# Description:
#   Bug 1966639 [Wayland] Use RequestAsyncRoundtrip/WaitForAsyncRoundtrips to get all events from registry r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D252140
# ==============================================================================

diff -r e0791cb06102 -r 5f5c42a58f8c widget/gtk/nsWaylandDisplay.cpp
--- a/widget/gtk/nsWaylandDisplay.cpp	Tue Jun 03 07:34:32 2025 +0000
+++ b/widget/gtk/nsWaylandDisplay.cpp	Tue Jun 03 07:34:32 2025 +0000
@@ -685,8 +685,9 @@
   mFormats = new DMABufFormats();
   mRegistry = wl_display_get_registry(mDisplay);
   wl_registry_add_listener(mRegistry, &registry_listener, this);
+  RequestAsyncRoundtrip();
   wl_display_roundtrip(mDisplay);
-  wl_display_roundtrip(mDisplay);
+  WaitForAsyncRoundtrips();
   EnsureDMABufFormats();
 
   for (auto& e : mSupportedTransfer) {
